core/test/CMakeLists.txt

```diff
@@ -3,54 +3,56 @@
 #############
 
 ## Add gtest based cpp test target and link libraries
-if (CATKIN_ENABLE_TESTING)
-	find_package(rostest REQUIRED)
+if (BUILD_TESTING)
+	find_package(ament_cmake_gtest REQUIRED)
+	find_package(ament_cmake_gmock REQUIRED)
 
-	add_library(gtest_utils gtest_value_printers.cpp models.cpp)
+	add_library(gtest_utils SHARED gtest_value_printers.cpp models.cpp)
 	target_link_libraries(gtest_utils ${PROJECT_NAME})
 
-	catkin_add_gtest(${PROJECT_NAME}-test-stage test_stage.cpp)
-	target_link_libraries(${PROJECT_NAME}-test-stage ${PROJECT_NAME} ${PROJECT_NAME}_stages gtest_utils gtest_main)
+	ament_add_gtest(${PROJECT_NAME}-test-stage test_stage.cpp)
```

Comment:
Are these already running and succeeding?

Reply:
Yes, they pass locally